src/GSoft.AspNetCore.Authentication.ClientCredentialsGrant/ClientCredentialsAuthorizeAttribute.cs

```diff
@@ -0,0 +1,22 @@
+﻿using GSoft.AspNetCore.Authentication.ClientCredentialsGrant;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.AspNetCore.Authorization;
+
+public sealed class ClientCredentialsAuthorizeAttribute : AuthorizeAttribute
+{
+    private readonly Dictionary<ClientCredentialsScope, string> _policyScopeMapping = new()
+    {
+        [ClientCredentialsScope.Read] = ClientCredentialsDefaults.AuthorizationReadPolicy,
+        [ClientCredentialsScope.Write] = ClientCredentialsDefaults.AuthorizationWritePolicy,
+        [ClientCredentialsScope.Admin] = ClientCredentialsDefaults.AuthorizationAdminPolicy,
+    };
+
+    public ClientCredentialsAuthorizeAttribute(ClientCredentialsScope scope)
+    {
+        this.Scope = scope;
+        this.Policy = this._policyScopeMapping.GetValueOrDefault(scope);
```

Comment:
Can we validate that the enum is actually part of the allowed values and throw otherwise?

Reply:
Yeah sure!